ui/src/pages/version/components/forms/components/AutoscalingPolicyFormGroup.js

```diff
@@ -0,0 +1,137 @@
+import React, { Fragment } from "react";
+import {
+  EuiFormRow,
+  EuiFieldNumber,
+  EuiDescribedFormGroup,
+  EuiSuperSelect,
+  EuiText,
+  EuiToolTip,
+  EuiIcon
+} from "@elastic/eui";
+
+export const AutoscalingPolicyFormGroup = ({
+  deploymentMode,
+  autoscalingPolicy,
+  onAutoscalingPolicyChange
+}) => {
+  const onMetricsTypeChange = value => {
+    onAutoscalingPolicyChange({
+      metrics_type: value,
+      target_value: autoscalingPolicy.target_value
+    });
+  };
+
+  const onTargetValueChange = e => {
+    onAutoscalingPolicyChange({
+      metrics_type: autoscalingPolicy.metrics_type,
+      target_value: e.target.value
+    });
+  };
+
+  const allMetricsType = [
+    {
+      value: "cpu_utilization",
+      inputDisplay: "CPU Utilization",
+      dropdownDisplay: (
+        <Fragment>
+          <strong>CPU Utilization</strong>
+          <EuiText size="s" color="subdued">
+            <p className="euiTextColor--subdued">
+              Autoscaling based on average cpu usage
+            </p>
+          </EuiText>
+        </Fragment>
+      )
+    },
+    {
+      value: "memory_utilization",
+      inputDisplay: "Memory Utilization",
+      dropdownDisplay: (
+        <Fragment>
+          <strong>Memory Utilization</strong>
+          <EuiText size="s" color="subdued">
+            <p className="euiTextColor--subdued">
+              Autoscaling based on average memory usage
+            </p>
+          </EuiText>
+        </Fragment>
+      )
+    },
+    {
+      value: "concurrency",
+      inputDisplay: "Concurrency",
+      dropdownDisplay: (
+        <Fragment>
+          <strong>Concurrency</strong>
+          <EuiText size="s" color="subdued">
+            <p className="euiTextColor--subdued">
+              Autoscaling based on number of concurrent request being processed
+            </p>
+          </EuiText>
+        </Fragment>
+      )
+    },
+    {
+      value: "rps",
+      inputDisplay: "Request per Second",
+      dropdownDisplay: (
+        <Fragment>
+          <strong>Request per Second</strong>
+          <EuiText size="s" color="subdued">
+            <p className="euiTextColor--subdued">
+              Autoscaling based on throughput (RPS)
+            </p>
+          </EuiText>
+        </Fragment>
+      )
+    }
+  ];
+
+  const filterMetricsOptions = (allOptions, deploymentMode) => {
+    var allowedMetrics = [
+      "cpu_utilization",
+      "memory_utilization",
+      "concurrency",
+      "rps"
+    ];
+    if (deploymentMode === "raw_deployment") {
+      allowedMetrics = ["cpu_utilization"];
```

Comment:
why only "cpu_utilization"? should we include "memory_utilization" ?

Reply:
memory utilization is not supported

Reply:
just curious why it is not supported for `raw_deployment` is it not supported by kserve?

Reply:
It's just not implemented yet (https://github.com/kserve/kserve/blob/13f4207f084bec8297a7eb055fa862e8b6f612f7/pkg/controller/v1beta1/inferenceservice/reconcilers/hpa/hpa_reconciler.go#L70) I think it can be added relatively easily if necessary.

Reply:
In that case for serverless it is not supported also?